porting/UpdatePortsFor2004.rst

```diff
@@ -0,0 +1,64 @@
+.. _UpdatePortsFor2004:
+
+Update ports for Ubuntu Touch 20.04
+===================================
+
+Ubuntu Touch 20.04 introduces a lot of changes underneath, which requires changes from the porters. This document is intended to serves as an overview on how to make your port runs on Ubuntu Touch 20.04. However, every port is different, and may requires more than what is outlined in this document. We'll be available in the UBports porting Telegram group to answer any question that might occur during process.
+
+For GitLab CI-based ports: switch to shared building script & GitLab CI configuration
+-------------------------------------------------------------------------------------
+
+Because of changes in how Ubuntu Touch 20.04's rootfs is made, we need to changes the way device tarballs are built. Instead of cherry-picking script changes from other port's repository, we recommend transitioning to the shared building script. By using the shared script, your port will receive the latest changes in device tarball building as rootfs continue to evolves (e.g. for 22.04 or newer). To transition to the shared script, remove the whole ``build/`` directory, and replace the content of ``build.sh`` with:
+
+.. code-block:: bash
+
+    #!/bin/bash
+    set -xe
+
+    [ -d build ] || git clone https://gitlab.com/ubports/community-ports/halium-generic-adaptation-build-tools -b halium-11 build
+    ./build/build.sh "$@"
+
+You may want to add ``build/`` into your ``.gitignore``.
+
+And for the same reason, we recommend using the shared GitLab CI configuration so that when the shared script has its requirement changes, your port will not be broken. Using the shared config can be done by replacing the content of ``.gitlab-ci.yml`` with:
+
+.. code-block:: yaml
+
+    include:
+        - https://gitlab.com/ubports/porting/community-ports/halium-generic-adaptation-build-tools/-/raw/halium-11/gsi-port-ci.yml
```

Comment:
Change halium-11 to main here so we can drop it eventually

Reply:
Done.

Reply:
And now that we mention it, note that this file should then also be updated to refer to `main` and use `main` branch by default.